Cannot build with versions >1.13.4

<!--
Thank you for reporting an issue.

Please fill in as much of the template below as you're able. Feel free to delete any section you want to skip.
-->

### Search terms you've used

polyfill, esbuild, undici

### Impacted package

Which packages do you think might be impacted by the bug ?

- [x] solid-client-authn-browser
- [ ] solid-client-authn-node
- [ ] solid-client-authn-core
- [ ] oidc-client-ext
- [ ] Other (please specify): ...

### Bug description

Since version 1.14 @inrupt/solid-client-authn-browser seems to depend on a lot of node internals. I am getting several errors when trying to build my app with esbuild, like these:

```
✘ [ERROR] Could not resolve "perf_hooks"

    ../node_modules/undici/lib/fetch/util.js:5:32:
      5 │ const { performance } = require('perf_hooks')
        ╵                                 ~~~~~~~~~~~~

  The package "perf_hooks" wasn't found on the file system but is built into node. Are you trying to
  bundle for node? You can use "platform: 'node'" to do that, which will remove this error.

✘ [ERROR] Could not resolve "assert"

    ../node_modules/undici/lib/fetch/body.js:19:23:
      19 │ const assert = require('assert')
         ╵                        ~~~~~~~~

  The package "assert" wasn't found on the file system but is built into node. Are you trying to
  bundle for node? You can use "platform: 'node'" to do that, which will remove this error.

✘ [ERROR] Could not resolve "util/types"

    ../node_modules/undici/lib/fetch/body.js:21:48:
      21 │ const { isUint8Array, isArrayBuffer } = require('util/types')

```

It worked fine with v1.13.4. Newer version seem to have introduced universal-fetch, which depends on undici, that causes the errors:

```
▶ npm ls undici   
  +-- @inrupt/solid-client-authn-browser@1.16.0
  | `-- @inrupt/universal-fetch@1.0.2
  |   `-- undici@5.22.1
```

The README only mentions to install the buffer polyfill, which I did. If other polyfills are required with the new version this should be documented.

### To Reproduce

1. Install esbuild and @inrupt/solid-client-authn-browser@1.16.0
2. Create file test.js:

```
import { login } from "@inrupt/solid-client-authn-browser";

function test() {
  login();
}

```

3. try to bundle it using esbuild: `esbuild --bundle test.js`

### Expected result

Successful build.

### Actual result

Errors shown above.

### Environment

Please run

npx envinfo --system --npmPackages --binaries --npmGlobalPackages --browsers

in your project folder and paste the output here:

```
$ npx envinfo --system --npmPackages --binaries --npmGlobalPackages --browsers
  System:
    OS: Linux 6.2 Debian GNU/Linux 11 (bullseye) 11 (bullseye)
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 4.02 GB / 31.05 GB
    Container: Yes
    Shell: 5.1.4 - /bin/bash
  Binaries:
    Node: 19.6.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 9.4.0 - /usr/local/bin/npm
  npmPackages:
    @inrupt/solid-client-authn-browser: 1.16.0 => 1.16.0 
    buffer: ^6.0.3 => 6.0.3 
    esbuild: ^0.18.11 => 0.18.11 
  npmGlobalPackages:
    corepack: 0.15.3
    npm: 9.4.0

```

## Additional information

<!-- Add any other relevant information that might be useful to understand and find a solution to the problem -->

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -23,10 +23,13 @@ The following have been deprecated, and will be removed in future major releases
 
 The following changes have been implemented but not released yet:
 
+## [1.17.0](https://github.com/inrupt/solid-client-authn-js/releases/tag/v1.17.0) - 2023-07-14
+
 ### New Feature
 
 - Support for [RP-Initiated Logout](https://openid.net/specs/openid-connect-rpinitiated-1_0.html) in Node and Browser
   libraries.
+- ESM support for Node and Browser packages.
 
 ## [1.16.0](https://github.com/inrupt/solid-client-authn-js/releases/tag/v1.16.0) - 2023-05-09
 

--- a/e2e/browser/test-app/package.json
+++ b/e2e/browser/test-app/package.json
@@ -1,6 +1,6 @@
 {
   "name": "test-app",
-  "version": "0.1.0",
+  "version": "1.17.0",
   "private": true,
   "scripts": {
     "dev": "next dev -p 3001",
@@ -10,7 +10,7 @@
   },
   "dependencies": {
     "@inrupt/internal-playwright-testids": "^2.0.2",
-    "@inrupt/solid-client-authn-browser": "^1.16.0",
+    "@inrupt/solid-client-authn-browser": "^1.17.0",
     "next": "^13.4.7",
     "react": "^18.2.0",
     "react-dom": "^18.2.0"

--- a/lerna.json
+++ b/lerna.json
@@ -1,4 +1,4 @@
 {
   "useNx": true,
-  "version": "1.16.0"
+  "version": "1.17.0"
 }

--- a/packages/browser/examples/demoAuthentiationMonitor/package.json
+++ b/packages/browser/examples/demoAuthentiationMonitor/package.json
@@ -1,7 +1,7 @@
 {
   "name": "demo-authentiation-monitor",
   "private": true,
-  "version": "1.0.0",
+  "version": "1.17.0",
   "description": "Demo for monitoring Authentication interactions in real-time",
   "main": "index.js",
   "scripts": {

--- a/packages/browser/examples/demoClientApp/package.json
+++ b/packages/browser/examples/demoClientApp/package.json
@@ -1,7 +1,7 @@
 {
   "name": "@inrupt/demo-app-authn-browser",
   "private": true,
-  "version": "1.0.0",
+  "version": "1.17.0",
   "description": "Demo client application for authentication in a browser",
   "main": "index.js",
   "scripts": {
@@ -30,7 +30,7 @@
     "webpack-dev-server": "^4.11.1"
   },
   "dependencies": {
-    "@inrupt/solid-client-authn-browser": "^1.16.0",
+    "@inrupt/solid-client-authn-browser": "^1.17.0",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "regenerator-runtime": "^0.13.11"

--- a/packages/browser/examples/single/bundle/package.json
+++ b/packages/browser/examples/single/bundle/package.json
@@ -1,7 +1,7 @@
 {
   "name": "browser-in-bundle",
   "private": true,
-  "version": "1.0.0",
+  "version": "1.17.0",
   "description": "",
   "main": "index.js",
   "scripts": {
@@ -25,9 +25,9 @@
     "webpack-dev-server": "^4.7.3"
   },
   "dependencies": {
+    "@inrupt/solid-client-authn-browser": "^1.17.0",
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
-    "regenerator-runtime": "^0.13.9",
-    "@inrupt/solid-client-authn-browser": "^1.16.0"
+    "regenerator-runtime": "^0.13.9"
   }
 }

--- a/packages/browser/examples/single/script/package.json
+++ b/packages/browser/examples/single/script/package.json
@@ -1,7 +1,7 @@
 {
   "name": "@inrupt/solid-client-authn-browser-via-script-tag-example",
   "private": true,
-  "version": "1.0.0",
+  "version": "1.17.0",
   "description": "",
   "main": "index.js",
   "scripts": {

--- a/packages/browser/package.json
+++ b/packages/browser/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@inrupt/solid-client-authn-browser",
-  "version": "1.16.0",
+  "version": "1.17.0",
   "license": "MIT",
   "types": "dist/index.d.ts",
   "browser": "dist/index.js",
@@ -29,8 +29,8 @@
     "@types/uuid": "^9.0.1"
   },
   "dependencies": {
-    "@inrupt/oidc-client-ext": "^1.16.0",
-    "@inrupt/solid-client-authn-core": "^1.16.0",
+    "@inrupt/oidc-client-ext": "^1.17.0",
+    "@inrupt/solid-client-authn-core": "^1.17.0",
     "@inrupt/universal-fetch": "^1.0.2",
     "events": "^3.3.0",
     "jose": "^4.3.7",

--- a/packages/core/package.json
+++ b/packages/core/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@inrupt/solid-client-authn-core",
-  "version": "1.16.0",
+  "version": "1.17.0",
   "license": "MIT",
   "main": "dist/index.js",
   "module": "dist/index.mjs",

--- a/packages/node/examples/bootstrappedApp/package.json
+++ b/packages/node/examples/bootstrappedApp/package.json
@@ -1,7 +1,7 @@
 {
   "name": "@inrupt/demo-authn-node-script",
   "private": true,
-  "version": "1.0.0",
+  "version": "1.17.0",
   "description": "Demo for script authentication in NodeJS",
   "main": "src/index.js",
   "scripts": {
@@ -18,8 +18,8 @@
   },
   "homepage": "https://github.com/inrupt/solid-client-authn-js#readme",
   "dependencies": {
+    "@inrupt/solid-client-authn-node": "^1.17.0",
     "express": "^4.17.3",
-    "yargs": "^16.1.1",
-    "@inrupt/solid-client-authn-node": "^1.16.0"
+    "yargs": "^16.1.1"
   }
 }

--- a/packages/node/examples/bootstrappedAppESM/package.json
+++ b/packages/node/examples/bootstrappedAppESM/package.json
@@ -1,12 +1,11 @@
 {
   "name": "@inrupt/demo-authn-node-script-esm",
   "private": true,
-  "version": "1.0.0",
+  "version": "1.17.0",
   "description": "Demo for script authentication in NodeJS",
   "main": "src/index.js",
   "type": "module",
-  "scripts": {
-  },
+  "scripts": {},
   "repository": {
     "type": "git",
     "url": "git+https://github.com/inrupt/solid-client-authn-js.git"
@@ -18,8 +17,8 @@
   },
   "homepage": "https://github.com/inrupt/solid-client-authn-js#readme",
   "dependencies": {
+    "@inrupt/solid-client-authn-node": "^1.17.0",
     "express": "^4.17.3",
-    "yargs": "^16.1.1",
-    "@inrupt/solid-client-authn-node": "^1.16.0"
+    "yargs": "^16.1.1"
   }
 }

--- a/packages/node/examples/demoClientApp/package.json
+++ b/packages/node/examples/demoClientApp/package.json
@@ -1,7 +1,7 @@
 {
   "name": "@inrupt/demo-app-authn-node",
   "private": true,
-  "version": "1.0.0",
+  "version": "1.17.0",
   "description": "Demo client application for authentication in NodeJS",
   "main": "src/index.js",
   "scripts": {
@@ -21,9 +21,9 @@
   },
   "homepage": "https://github.com/inrupt/solid-client-authn#readme",
   "dependencies": {
+    "@inrupt/solid-client-authn-node": "../../",
     "express": "^4.17.1",
-    "loglevel": "^1.7.0",
-    "@inrupt/solid-client-authn-node": "../../"
+    "loglevel": "^1.7.0"
   },
   "devDependencies": {
     "@inrupt/eslint-config-base": "0.0.4",

--- a/packages/node/examples/multiSession/package.json
+++ b/packages/node/examples/multiSession/package.json
@@ -1,7 +1,7 @@
 {
   "name": "@inrupt/demo-authn-node-script-multisession",
   "private": true,
-  "version": "1.0.0",
+  "version": "1.17.0",
   "description": "Demo for session management in NodeJS",
   "main": "src/index.js",
   "scripts": {
@@ -18,7 +18,7 @@
   },
   "homepage": "https://github.com/inrupt/solid-client-authn-js#readme",
   "dependencies": {
-    "@inrupt/solid-client-authn-node": "^1.16.0",
+    "@inrupt/solid-client-authn-node": "^1.17.0",
     "cookie-session": "^1.4.0",
     "express": "^4.17.1"
   },

--- a/packages/node/package.json
+++ b/packages/node/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@inrupt/solid-client-authn-node",
-  "version": "1.16.0",
+  "version": "1.17.0",
   "license": "MIT",
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
@@ -27,7 +27,7 @@
     "@types/uuid": "^9.0.1"
   },
   "dependencies": {
-    "@inrupt/solid-client-authn-core": "^1.16.0",
+    "@inrupt/solid-client-authn-core": "^1.17.0",
     "@inrupt/universal-fetch": "^1.0.1",
     "jose": "^4.3.7",
     "openid-client": "~5.4.2",

--- a/packages/oidc-browser/package.json
+++ b/packages/oidc-browser/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@inrupt/oidc-client-ext",
-  "version": "1.16.0",
+  "version": "1.17.0",
   "description": "A module extending oidc-client-js with new features, such as dynamic client registration and DPoP support.",
   "homepage": "https://github.com/inrupt/solid-client-authn-js/tree/main/packages/oidc/",
   "bugs": "https://github.com/inrupt/solid-client-authn-js/issues",
@@ -22,7 +22,7 @@
   },
   "dependencies": {
     "@inrupt/oidc-client": "^1.11.6",
-    "@inrupt/solid-client-authn-core": "^1.16.0",
+    "@inrupt/solid-client-authn-core": "^1.17.0",
     "@inrupt/universal-fetch": "^1.0.1",
     "jose": "^4.10.0",
     "uuid": "^9.0.0"

